edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Hosting.cs

```diff
@@ -31,8 +35,46 @@ public class Hosting
             bool clientCertAuthEnabled,
             SslProtocols sslProtocols)
         {
+            /* Define the HttpsConnectionAdapterOptions which integrates with aspnetcore https middleware to help us manage the client cert chain.
+               We need to store the client cert chain in the connection context, so that we can perform custom CA validation later to authenticate with the service identity.
+               We are using logic here that depends on an undocumented api, however eliminating this dependency would require a large refactor.
+               Context in this issue: https://github.com/dotnet/aspnetcore/issues/21606 */
```

Comment:
Give link to httpsmiddleware:

https://github.com/dotnet/aspnetcore/blob/5a0c097ad4b0aef756cd92f2d593cda3fa223748/src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs